vulnerable_people_form/integrations/submission_tracing.py

```diff
@@ -0,0 +1,80 @@
+import json
+import time
+import hashlib
+import random
+import logging
+
+from flask import request, current_app
+
+from vulnerable_people_form.form_pages.shared.logger_utils import init_logger, log_event_names, create_log_message
+
+logger = logging.getLogger(__name__)
+init_logger(logger)
+
+
+def _client_ip():
+    if request:
+        if request.environ.get("HTTP_X_FORWARDED_FOR") is None:
+            return request.environ["REMOTE_ADDR"]
+        else:
+            # if behind a proxy
+            return request.environ["HTTP_X_FORWARDED_FOR"]
+    return None
+
+
+def _user_agent():
+    result = {"platform": None, "browser": None, "version": None, "string": None}
+    if request:
+        ua_obj = request.user_agent
+        result["platform"] = ua_obj.platform
+        result["browser"] = ua_obj.browser
+        result["version"] = ua_obj.version
+        result["string"] = ua_obj.string
+    return result
+
+
+def _peppering(obj):
+    if obj is not None and current_app:
+        pepper = current_app.config.get("SUBMISSION_TRACING_PEPPER")
```

Comment:
i don't think this check is required and the pepper should be passed into this method. The main method below handles the validation of the SUBMISSION_TRACING_PEPPER and as this is an internal method it should never be called anywhere else.


  ```
def _peppering(obj, pepper):
    m = hashlib.sha256()
    # convert to string and lower case
    objstr = str(obj).lower()
    # get only the alphanumeric characters
    objstr = "".join(e for e in objstr if e.isalnum())
    # generate hash
    m.update(f"{pepper}:{objstr}".encode())
    # return first 32 characters of the 64 character hash hex
    return m.hexdigest()[:32]
    
```